clang/test/CIR/Lowering/OpenMP/master.cir

```diff
@@ -0,0 +1,29 @@
+// RUN: cir-translate %s -cir-to-llvmir | FileCheck %s
+
+!s32i = !cir.int<s, 32>
+module {
+  cir.func @omp_master_test() {
+    %y = cir.alloca !s32i, !cir.ptr<!s32i>, ["y", init] {alignment = 4 : i64}
+    %y_val = cir.const #cir.int<0> : !s32i
+    cir.store %y_val, %y : !s32i, !cir.ptr<!s32i>
+
+    // CHECK-LABEL: @omp_master_test
+    // CHECK: call i32 @__kmpc_global_thread_num
+    // CHECK: call i32 @__kmpc_master({{.*}},{{.*}})
```

Comment:
Are these already tested as part of the openmp dialect? If so we probably don't need to test them again for each feature you add. As CIR changes we might need to keep updating this file, which is probably non-ideal if things are already tested elsewhere.

An alternative option is to add one more `RUN` line to `clang/test/CIR/CodeGen/OpenMP/master.cpp` and add a `emit-llvm` instead of `emit-cir`,  you can then test `@__kmpc_global_thread_num` and others there directly from source code - we do similar for CIR/LLVM in `clang/test/CIR/CodeGen/abstract-cond.c`, you can get some inspiration there.

Reply:
I'm guessing some of these are already being tested on MLIR and that CIR might change, fair, but then what is the purpose of this lowering tests? If there were changes in CIR then it would affect other tests not just this one, take for instance ``` clang/test/CIR/Lowering/ternary.cir ```. 

I could change the CodeGen ones similar to the abstract-cond.c, but does that mean that I shouldn't make more lowering tests for new features? Thank you!

Reply:
> I'm guessing some of these are already being tested on MLIR and that CIR might change, fair, but then what is the purpose of this lowering tests?

Pre-existing tests are one thing, adding new tests is another, I'm asking you to test the LLVM output altogether with the source code, there's no good reason in this PR to add a `.cir` test that is testing the same thing. Note that you are only changing CIRGen, not really lowering. Btw, this isn't a strict rule, but I'd prefer to keep CIR to CIR tests to more minimal testcases because of maintainance burden. In this specific PR I don't see any reason why a CIR to CIR test improves anything.

> If there were changes in CIR then it would affect other tests not just this one, take for instance clang/test/CIR/Lowering/ternary.cir.

Yes, and probably most of them, and we have to go and change it all.

> I could change the CodeGen ones similar to the abstract-cond.c, but does that mean that I shouldn't make more lowering tests for new features? Thank you!

It's on a per case basis, like I said before, not sure how this CIR test helps anything here. You just need to add one more RUN line to master.cpp and check for the same stuff.